overlord/snapstate/check_snap.go

```diff
@@ -596,6 +596,19 @@ func earlyEpochCheck(info *snap.Info, snapst *SnapState) error {
 	return checkEpochs(nil, info, cur, nil, Flags{}, nil)
 }
 
+func earlyChecks(st *state.State, snapst *SnapState, update *snap.Info, flags Flags) (Flags, error) {
+	var err error
+	flags, err = ensureInstallPreconditions(st, update, flags, snapst)
```

Comment:
Silly question, why not just `flags, err := ensureInstallPre...` here and remove the `var err error` above?